tests/reftests/working-dir.test

```diff
@@ -234,3 +234,46 @@ test "-f" "present" (CWD=${BASEDIR}/OPAM/working-dir/.opam-switch/build/qux.4)
 -> compiled  qux.4
 -> installed qux.4
 Done.
+### : Shows that opam correctly handles updated constraints with --working-dir
+### : See https://github.com/ocaml/opam/issues/5178
+### opam pin remove ./ongoing
+Ok, ongoing is no longer pinned to git+file://${BASEDIR}/ongoing#master (version dev)
+The following actions will be performed:
+=== remove 1 package
+  - remove ongoing dev
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> removed   ongoing.dev
+Done.
+### <pkg:qux.1>
+opam-version:"2.0"
+### <pin:ongoing/ongoing.opam>
+opam-version: "2.0"
+depends: ["qux"]
+### opam install ./ongoing --working-dir
+Package ongoing does not exist, create as a NEW package? [y/n] y
+ongoing is now pinned to git+file://${BASEDIR}/ongoing#master (version dev)
+[ongoing.dev] synchronised (git+file://${BASEDIR}/ongoing#master)
+The following actions will be performed:
+=== install 1 package
+  - install ongoing dev (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed ongoing.dev
+Done.
+### <pin:ongoing/ongoing.opam>
+opam-version: "2.0"
+depends: ["qux" {= "1"}]
+### opam install ./ongoing --working-dir --show-action
+[NOTE] Package ongoing is already installed (current version is dev).
```

Comment:
In my opinion it’s a bit surprising that `opam install` behaves like this on local packages (or even for "global" ones) when they have changed. But this is a discussion for another PR and isn’t a blocker for this one.

Reply:
Related to https://github.com/ocaml/opam/issues/5059 maybe?

Reply:
The check is done afterwards. But agree, for working-dir maybe we should check before if there was changes.

It is not directly related to #5059, it's different mechanisms.